### ISSUE TITLE
JoinLayers() eliminates metadata

### ISSUE THREAD

**Issue by jpvoth** [satijalab/seurat#8207]
Hi - I updated to Seurat v5 from v4.3 and have been trying to go through the vignettes on data integration and clustering. I had this figured out for v4.3, but the v5 has thrown me for a loop thus far. 
My steps:
1. convert 33 h5 samples to seurat objects with metadata attached - worked fine
2. merge 33 seurat objects to one big object - worked, but created 33 layers under [["RNA"]] 
3. Use JoinLayers() to merge all these layers together - these layers are not useful so want to make meaningful layers.
--- this is where the issue arises --- this joining process has wiped the metadata - essentially making any downstream analysis impossible. Another note - this converted Seurat Object to a new type "Large Assay5", which I have never seen but understand is the Seurat Object format. However - to actually integrate this data I need to get rid of these layers to make meaningful ones. 

How do I merge these layers while preserving metadata? IntegrateLayers() may be what I need, but I'm trying to use IntegrateLayers() for correcting for sequencing batch effects, NOT individual sample effects. The current layers are for each individual sample.

**Comment by jpvoth** [comment]
```
setwd("~/individual.seurats/")
files <- list.files(".")
first <- readRDS(file ="~/seurat1.rds" )
second <- readRDS(file= "~/seurat2.rds")
seurat <- merge(first, y=second, project = "Final.comp", merge.data=TRUE)

for (i in files){
temp <- readRDS(i) 
seurat <- merge(seurat, y=temp, project = "Final.comp",  merge.data=TRUE)  
}
```
_______________________________________________________________________________________________________________________
Run that block - then this:
```
md <-seurat@meta.data
md %>% count(orig.ident)
```
-----> confirms all merged individual seurats have right number of cells by checking metadata - works. Also shows the 33 layers at this point.

`seurat <- JoinLayers(seurat[["RNA"]]
`
--> this creates "Large Assay5" and metadata is no longer retrievable.

**Comment by jpvoth** [comment]
![image](https://github.com/satijalab/seurat/assets/150742581/2ee12e77-b0d8-4830-ae17-5ae9230e9b4b)

**Comment by jpvoth** [comment]
Interestingly - I had previously merged these files into a big seurat in v4.3, and I tried to split that combined file using:

`seurat[["RNA"]] <- split(seurat[["RNA"]], f=seurat$orig.ident)`

And that actually worked fine. Metadata is preserved, new layers are present (although it created "counts.seurat1........" and "data.seurat1...................." layers for each. Not sure if that is to be expected or not. So - the issue seems to be specifically on the JoinLayers() function. I remerge the layers I had just created and it wiped the metadata again.
